### DISCUSSION TITLE
Save changes in PixelCollection to image?

### DISCUSSION THREAD

**Discussion by hunterlan** [dlemstra/Magick.NET#941] [Help]
Hi! 

I'm writing realization of LSB(Least Significant Bit) by using Magick.NET and I faced with problem, how to save changes, which was done in PixelCollection.

It's my code:

            public bool EncodeTextToImage(string text, string path)
            {
            int length = Math.Min(text.Length, 255);
            MagickImage image = null;
            try
            {
                image = new MagickImage(path);
            }
            catch
            {
                return false;
            }
            
            for (int i = 0; i < ONE_BYTE; i++)
            {
                var pixels = image.GetPixels();
                var color = pixels[i, image.Height - 1].ToColor();
                byte red = color.R;

                red = Convert.ToByte( (red & 254) | ((length & (1 << i)) > 0 ? 1 : 0) );
                var newColors = new MagickColor
                {
                    A = color.A,
                    B = color.B,
                    G = color.G,
                    K = color.K,
                    R = red
                };
                pixels.SetPixel(new Pixel(i, image.Height - 1, newColors.ToByteArray()));
            }

            int x = ONE_BYTE;
            int y = image.Height - 1;
            for (int i = 0; i < length; i++)
            {
                int character = text[i];
                for (int j = 0; j < ONE_BYTE; j++)
                {
                    if (x >= image.Width)
                    {
                        y--;
                        x = 0;
                    }
                    var pixels = image.GetPixels();
                    var color = pixels[x, y].ToColor();
                    byte red = color.R;

                    red = Convert.ToByte( (red & 254) | ((character & (1 << i)) > 0 ? 1 : 0) );
                    var newColors = new MagickColor
                    {
                        A = color.A,
                        B = color.B,
                        G = color.G,
                        K = color.K,
                        R = red
                    };
                    pixels.SetPixel(new Pixel(i, image.Height - 1, newColors.ToByteArray()));
                    x++;
                }
            }

            string[] result_path = path.Split('.');
            result_path[0] = string.Concat(result_path[0], "_resultEncrypting.");
            try
            {
                image.Write(string.Concat(result_path));
            }
            catch
            {
                return false;
            }
            return true;
        } 
 
When I set pixels, this changes doesn't apply to variable "image". I tried to assign new MagickImage to "image", and pass to constructor array of byte, but, it throws error (different errors, depends on which constructor I will choose).

So, how can I save it?

**Comment by dlemstra** [comment]
Why do you think that the pixels are not changed? Your sample does not include enough information for me to test your code on my machine.

p.s. you should `.Dispose() / using` the `image` and the `pixels`.

**Reply by hunterlan** [reply]
Sorry, for long answer. 

I'm not sure, that I wrote my code correctly. I will put down all source code. 

[LSB.cs](https://pastebin.com/3C6652WA)
[Program.cs](https://pastebin.com/EThbnQax)
